File: classes/parte-6-datos-y-estructuras/100-enumeraciones-y-tipos-algebraicos-adt-sum-types/implementaciones/java/Main.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] t = br.readLine().trim().split("\\s+");
        long area;
        if (t[0].equals("cuadrado")) {
            long l = Long.parseLong(t[1]);
            area = l * l;
        } else {
            area = Long.parseLong(t[1]) * Long.parseLong(t[2]);
        }
        System.out.println("area=" + area);
    }
}

File: classes/parte-6-datos-y-estructuras/100-enumeraciones-y-tipos-algebraicos-adt-sum-types/implementaciones/typescript/main.ts
import { readFileSync } from "node:fs";

const t: string[] = readFileSync(0, "utf8").trim().split(/\s+/);
let area: number;
if (t[0] === "cuadrado") area = Number(t[1]) ** 2;
else area = Number(t[1]) * Number(t[2]);
console.log(`area=${area}`);
